src/librepgp/stream-parse.cpp

```diff
@@ -2107,8 +2107,17 @@ init_encrypted_src(pgp_parse_handler_t *handler, pgp_source_t *src, pgp_source_t
             }
             /* Decrypt key */
             rnp::KeyLocker seclock(*seckey);
-            if (!seckey->unlock(*handler->password_provider, PGP_OP_DECRYPT)) {
-                errcode = RNP_ERROR_BAD_PASSWORD;
+            int            attempts = 3;
+            while (attempts--) {
+                errcode = RNP_ERROR_NO_SUITABLE_KEY;
+                if (!seckey->unlock(*handler->password_provider, PGP_OP_DECRYPT)) {
```

Comment:
Here and below we should take in account case when there are multiple recipients of the message, and user want to enter password only to the particular one. This may be checked by returning result of password provider call - false or true.

Reply:
Also we should make number of password request tries configurable.

Reply:
> This may be checked by returning result of password provider call - false or true.

In case of a `rnp` cli tool and password provider that reads from the `pass-fd`, it does not check the keyid, so the password and `true` result  would be returned every time.

Reply:
Otherwise this code should still succeed when one of the passwords matches one of the keys, just many "Bad password" messages will be printed on stderr as a side effect.

Reply:
> In case of a `rnp` cli tool and password provider that reads from the `pass-fd`, it does not check the keyid, so the password and `true` result would be returned every time.

These changes are mostly not for CLI, but for other FFI users (like TB), to let them repeat the password request if needed.